app/Exceptions/NoExchangesCanBeDone.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace App\Exceptions;
+
+use Exception;
+
+class NoExchangesCanBeDone extends Exception
```

Comment:
```suggestion
class NoExchangesCanBeDoneException extends Exception
```

I think saying in the class name that it is an exception makes it much clearer